source/renderer/app/components/loading/StatusIcons.js

```diff
@@ -0,0 +1,257 @@
+  },
+  checkingIfNodeIsSubscribed: {
+    id: 'status.icons.checkingIfNodeIsSubscribed',
+    defaultMessage: 'Checking if Node is subscribed!',
+    description:
+      'Message "Checking if Node is subscribed" on the status icon tooltip',
+  },
+  nodeTimeIsCorrect: {
+    id: 'status.icons.nodeTimeIsCorrect',
+    defaultMessage: 'Node time is correct!',
+    description: 'Message "Node time is correct" on the status icon tooltip',
+  },
+  nodeTimeIsNotCorrect: {
+    id: 'status.icons.nodeTimeIsNotCorrect',
+    defaultMessage: 'Node time is not correct!',
+    description:
+      'Message "Node time is not correct" on the status icon tooltip',
+  },
+  checkingIfNodeTimeIsCorrect: {
+    id: 'status.icons.checkingIfNodeTimeIsCorrect',
+    defaultMessage: 'Checking if Node time is correct!',
+    description:
+      'Message "Checking if Node time is correct" on the status icon tooltip',
+  },
+  nodeIsSyncing: {
+    id: 'status.icons.nodeIsSyncing',
+    defaultMessage: 'Node is syncing!',
+    description: 'Message "Node is syncing" on the status icon tooltip',
+  },
+  nodeIsNotSyncing: {
+    id: 'status.icons.nodeIsNotSyncing',
+    defaultMessage: 'Node is not syncing!',
+    description: 'Message "Node is not syncing" on the status icon tooltip',
+  },
+  checkingIfNodeIsSyncing: {
+    id: 'status.icons.checkingIfNodeIsSyncing',
+    defaultMessage: 'Checking if Node is syncing!',
+    description:
+      'Message "Checking if Node is syncing" on the status icon tooltip',
+  },
+});
+
+type Props = {
+  nodeState: ?CardanoNodeState,
+  isNodeResponding?: boolean,
+  isNodeSubscribed?: boolean,
+  isNodeTimeCorrect?: boolean,
+  isNodeSyncing?: boolean,
+};
+
+const STATUS_CLASSNAMES: Object = {
+  [CardanoNodeStates.STARTING]: 'unloaded',
+  [CardanoNodeStates.RUNNING]: 'on',
+  [CardanoNodeStates.EXITING]: 'unloaded',
+  [CardanoNodeStates.STOPPING]: 'unloaded',
+  [CardanoNodeStates.STOPPED]: 'unloaded',
+  [CardanoNodeStates.UPDATING]: 'unloaded',
+  [CardanoNodeStates.UPDATED]: 'unloaded',
+  [CardanoNodeStates.CRASHED]: 'off',
+  [CardanoNodeStates.ERRORED]: 'off',
+  [CardanoNodeStates.UNRECOVERABLE]: 'off',
+  true: 'on',
+  false: 'off',
+  undefined: 'unloaded',
+};
+
+const MESSAGES = {
+  [CardanoNodeStates.RUNNING]: messages.nodeIsRunning,
+  [CardanoNodeStates.STARTING]: messages.nodeIsStarting,
+  [CardanoNodeStates.EXITING]: messages.nodeIsExiting,
+  [CardanoNodeStates.STOPPING]: messages.nodeIsStopping,
+  [CardanoNodeStates.STOPPED]: messages.nodeHasStopped,
+  [CardanoNodeStates.UPDATING]: messages.nodeIsUpdating,
+  [CardanoNodeStates.UPDATED]: messages.nodeHasBeenUpdated,
+  [CardanoNodeStates.CRASHED]: messages.nodeHasCrashed,
+  [CardanoNodeStates.ERRORED]: messages.nodeHasErrored,
+  [CardanoNodeStates.UNRECOVERABLE]: messages.nodeIsUnrecoverable,
+  nodeStatenull: messages.nodeIsStarting,
+  isNodeRespondingtrue: messages.nodeIsResponding,
```

Comment:
Please continue with camelCase for the variable types … e.g: `isNodeRespondingTrue`

Reply:
I've simplified the logic, so it accesses the messages more easily.